src/Models/HydrostaticFreeSurfaceModels/implicit_free_surface.jl

```diff
@@ -25,16 +25,15 @@ ImplicitFreeSurface(; gravitational_acceleration=g_Earth) =
 
 # Internal function for HydrostaticFreeSurfaceModel
 function FreeSurface(free_surface::ImplicitFreeSurface{Nothing}, velocities, arch, grid)
-    #η = CenterField(arch, grid, TracerBoundaryConditions(grid))
-    η = ReducedField(Center, Center, Nothing, arch, grid; dims=3)
+    η = FreeSurfaceDisplacementField(velocities, arch, grid)
     g = convert(eltype(grid), free_surface.gravitational_acceleration)
 
-    barotropic_x_volume_flux = ReducedField(Face, Center, Nothing, arch, grid; dims=(3), boundary_conditions=nothing)
-    barotropic_y_volume_flux = ReducedField(Center, Face, Nothing, arch, grid; dims=(3), boundary_conditions=nothing)
+    barotropic_x_volume_flux = ReducedField(Face, Center, Nothing, arch, grid; dims=3)
```

Comment:
I don't know exactly how `ReducedField` is defined but if we have `nothing` in the `z` direction, then is the number of dimensions still 3? I guess it's a matter of convention and just trying to understand what's used here.

Reply:
We use `Nothing` to represent a "reduced" location (eg, if a field is reduced along a particular dimension, it does not have a location in that dimension). It's actually not necessary to explicitly specify a `Nothing` location because locations along reduced dimensions are automatically converted to `Nothing` inside the `ReducedField` constructor.

The underlying data for a `ReducedField` is always three dimensional --- but reduced dimensions are singleton; or have a length of 1.

Reply:
@glwagner what do you think about renaming dims() => dimstoreduce()? When I first used this it was unclear whether I should specify dims to keep or dims to reduce.

Reply:
We could use `reduced_dims` instead of `dims`. I used `dims` to match the syntax for other reduction operations in julia (`mean`, `std`, `maximum`, etc). I think I see your point though. It'd be good to have another opinion, @ali-ramadhan ?

Reply:
I agree that `reduced_dims` is clearer than just `dims` but yeah I think we've been trying to stick to base Julia style here where `dims` is always the "dimensions to which the function is applied", e.g.
```
help?> mean

  mean(f::Function, A::AbstractArray; dims)

  Apply the function f to each element of array A and take the mean over dimensions dims.
```

I guess I would vote for `dims` here since using `reduced_dims` would suggest we should use e.g. `averaged_dims` for `AveragedField`. I think being consistent with base Julia usage of `dims` is hopefully clear enough.

Reply:
Thanks for the discussion and I'm happy with using `dim`, and now it makes more sense because of the context.